core/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -37,11 +39,20 @@
 
     private final PipelineStore pipelineStore;
     private final PipelineExecutionService pipelineExecutionService;
-    private final ProcessorsRegistry.Builder processorsRegistryBuilder;
 
-    public IngestService(Settings settings, ThreadPool threadPool, ProcessorsRegistry.Builder processorsRegistryBuilder) {
-        this.processorsRegistryBuilder = processorsRegistryBuilder;
-        this.pipelineStore = new PipelineStore(settings);
+    public IngestService(Settings settings, ThreadPool threadPool,
+                         Environment env, ScriptService scriptService, List<IngestPlugin> ingestPlugins) {
+        final TemplateService templateService = new InternalTemplateService(scriptService);
+        Map<String, Processor.Factory> processorFactories = new HashMap<>();
+        for (IngestPlugin ingestPlugin : ingestPlugins) {
+            Map<String, Processor.Factory> newProcessors = ingestPlugin.getProcessors(env, scriptService, templateService);
+            for (Map.Entry<String, Processor.Factory> entry : newProcessors.entrySet()) {
+                if (processorFactories.put(entry.getKey(), entry.getValue()) != null) {
+                    throw new IllegalArgumentException("Ingest processor [" + entry.getKey() + "] is already registered");
```

Comment:
maybe add a small test for this? (there was a small unit test for this: ProcessorsRegistryTests)